Anaconda e SSH

J-Linaris opened the pull request:
Adição do tutorial da instalação e uso do anaconda junto ao tutorial do acesso com ssh. Além disso, foi comentada a parte de configuração do arquivo .bashrc

guilherme-dna committed:
Added SSH access tutorial

guilherme-dna committed:
update in SSH tutorial

guilherme-dna committed:
minor spelling errors fixes

guilherme-dna committed:
minor fixes and additional info about keygen

J-Linaris committed:
adiciona o tutorial de instalacao e configuracao do anaconda

J-Linaris committed:
Merge branch 'config-env' of https://github.com/lucasmsobrinho/docs-lab-vision into config-env

J-Linaris committed:
Correcao de conflitos entre commits

J-Linaris committed:
<usuario_vision> virou <user_science> para manter o padrao e adicao da seçao 'Permissoes e acessos'.

J-Linaris committed:
padronizacao com Rede e-Science, mudanca da ordem dos tutoriais e mudanca do titulo do tutorial do ssh

J-Linaris committed:
comenta a parte de bashrc

guilherme-dna committed:
Fixing error in scp command line

fixed missing `:~` in scp command for sending archives

J-Linaris committed:
arrumando o tutorial de configuracao do anaconda

J-Linaris committed:
merge
Merge branch 'config-env' of https://github.com/lucasmsobrinho/docs-lab-vision into config-env

lucasmsobrinho commented on mkdocs.yml:
```diff
@@ -1,5 +1,5 @@
 site_name: Documentação Rede Vision
-site_url: https://vision.ime.usp.br/~lucasmsobrinho
+site_url: https://vision.ime.usp.br/~jplinaris
```
Essa configuração é só para sua máquina. Não deve ser enviada.

lucasmsobrinho commented on docs/tutorials/configuring-workplace.md:
```diff
@@ -1,26 +1,209 @@
+Para a Rede e-Science, temos um interesse especial no Anaconda tendo em vista que o usuário comum, ou seja, um usuário que não seja administrador, não tem acesso ao comando sudo, essencial para instalar pacotes. Sendo assim, ao usar Anaconda, contornamos o problema de não se ter permissão para executar o comando sudo.
+
+### Instalando o anaconda no meu usuário da Rede e-Science
+Como primeiro passo, baixe, no seu computador, a versão mais recente do anaconda disponível neste repositório:
```
Dá pra fazer esse download direta na sua máquina vision usando o comando `wget`:

```bash
wget https://repo.anaconda.com/archive/Anaconda3-2024.10-1-Linux-x86_64.sh
```

lucasmsobrinho commented on docs/tutorials/configuring-workplace.md:
```diff
@@ -1,26 +1,209 @@
+
+E assim, você estará conectado à máquina de entrada ( lembra que nunca se deve fazer experimentos nela! ).
+
+## 2. Anaconda 
```
Não sei se eu particularmente concordo com a indicação Anaconda, pois pode usar muito do seu espaço da vision com dezenas de pacotes que você não vai usar. Lembrando que existe a cota de 100GB na pasta home.

lucasmsobrinho commented on docs/tutorials/configuring-workplace.md:
```diff
@@ -1,26 +1,209 @@
+
+Anaconda é uma plataforma para se desenvolver projetos em python e R. Assim, dispõe de mais de 7500 pacotes adicionais, que podem ser instalados pelo repositório do Anaconda, fora os 300 pacotes que já vem instalados ao baixá-lo. De forma simples, o Anaconda permite a criação de ambientes de programação diversos e independentes entre si, o que facilita o desenvolvimento de programas que dependem  ...
+
+https://en.wikipedia.org/wiki/Anaconda_(Python_distribution)
```
É interessante usar o [guia de estilo markdown do google](https://google.github.io/styleguide/docguide/style.html#use-informative-markdown-link-titles).

Ficaria algo como:
```recomendamos a página do [Anaconda na Wikipedia](url)```

Ainda melhor que o Wikipedia seria a [própria página de documentação do Anaconda](https://docs.anaconda.com/).

lucasmsobrinho commented on docs/tutorials/configuring-workplace.md:
```diff
@@ -1,26 +1,209 @@
+
+https://en.wikipedia.org/wiki/Anaconda_(Python_distribution)
+
+Para a Rede e-Science, temos um interesse especial no Anaconda tendo em vista que o usuário comum, ou seja, um usuário que não seja administrador, não tem acesso ao comando sudo, essencial para instalar pacotes. Sendo assim, ao usar Anaconda, contornamos o problema de não se ter permissão para executar o comando sudo.
```
Acho que isso cabe na introdução, na recomendação pra usar o conda.

lucasmsobrinho commented on docs/tutorials/configuring-workplace.md:
````diff
@@ -1,26 +1,209 @@
+```bash
+eval "$(/home/user1/anaconda3/bin/conda shell.bash hook)"
+```
+Quando entrar no ambiente base do anaconda, ou seja, quando o seu terminal parecer algo como `(base)user_science@net03:~$ `, rode o comando:
````
Acho que não é recomendado fazer nada nessa máquina de entrada shell, nem instalações.
Muda esse net03 para qualquer outra coisa

lucasmsobrinho commented on docs/tutorials/configuring-workplace.md:
````diff
@@ -1,26 +1,209 @@
+A fim de configurar o anaconda, será necessário ativá-lo pela primeira vez, o que é feito a partir da execução do seguinte comando:
+
+```bash
+eval "$(<local_instalacao_anaconda>/conda shell.YOUR_SHELL_NAME hook)"
````
Não entendi isso direito, é a mesma coisa que que rodar ```./anaconda3/conda``` ?

J-Linaris replied:
é que, como primeira configuração, depois de instalar o miniconda ou o anaocnda3, ambos pedem para rodar esse eval para ativá-lo pela primeira vez, aí você roda, quando entrar no ambiente base, conda init para terminar a configuração do anaconda.

lucasmsobrinho commented on docs/tutorials/configuring-workplace.md:
````diff
@@ -1,26 +1,209 @@
 
-### 1. Acesso ao Servidor via SSH
+```bash
+(myenv)user_science@deepone:~$ conda install numpy
````
Deixa só o comando para execução.

lucasmsobrinho reviewed (changes requested):
Bom trabalho. O tutorial tem bastante informação e indicações claras.
Adicioneis algumas correções pequenas pra fazer e a questão do anaconda vs miniconda.

Pensando melhor também em termos de organização, acho que essa parte do SSH cabe em outro arquivo separado.

Ficando os arquivos:
access.md
configuring-worplace.md -> installing-python.md

guilherme-dna committed:
fixing site_url config

from ~jplinaris to ~lucasmsobrinho

guilherme-dna committed:
removing non-command part from bash example

removing (myenv) from a command to make it work from just copy-paste

J-Linaris committed:
new file for ssh tutorial and requested changes in anaconda tutorial

J-Linaris committed:
muda 'máquina base' por 'máquina de acesso'

lucasmsobrinho commented on docs/tutorials/installing-python.md:
```diff
@@ -0,0 +1,144 @@
+Em relação à Rede e-Science, temos um interesse especial no Miniconda, uma versão do Anaconda que não tem nenhum pacote pré-instalado, de forma que ocupa muito menos espaço que o anaconda3 e apresenta as mesmas funcionalidades.
+
+### Instalando o miniconda no meu usuário da Rede e-Science
+Como primeiro passo, baixe, no seu computador, a versão mais recente do miniconda a partir da execução do seguinte comando:
```
Você consegue baixar o miniconda3 usando wget direto na máquina da vision.

Daí não precisa fazer o scp. (remover linhas 21-36) e 
adaptar o texto da linha 36 para algo como:
!!! warning "Atenção"
     Nunca use a máquina de entrada para executar processos que usam muitos recursos. Faça a instalação em outras máquinas (deeps, russas, cientistas, ...).

J-Linaris committed:
merge
Merge branch 'master' of https://github.com/lucasmsobrinho/docs-lab-vision into config-env

J-Linaris committed:
muda o tutorial para, ao invés de instalar o miniconda na local e passar para a vision, orienta a baixar direto na vision

guilherme-dna committed:
adding ssh tunneling e minor fixes

J-Linaris committed:
Merge branch 'master' of https://github.com/lucasmsobrinho/docs-lab-vision into config-env

J-Linaris committed:
merge Merge branch 'config-env' of https://github.com/lucasmsobrinho/docs-lab-vision into config-env

victor-nasc commented on docs/tutorials/installing-python.md:
````diff
@@ -0,0 +1,138 @@
+Agora, para instalar de fato o miniconda, devemos executar o bash script anteriormente baixado a partir do seguinte comando:
+
+```bash
+bash Miniconda3-latest-Linux-x86_64.sh
````
Nao precisa dar `$chmod +x Miniconda3-latest-Linux-x86_64.sh` antes?

J-Linaris replied:
Também tive essa dúvida, porém, testando, vi que não é necessário.

J-Linaris replied:
Apesar de ter funcionado sem conceder permissão para executar antes de rodar o comando bash Miniconda..., mudei o tutorial para incluir esse comando que concede permissão para execução do script, só para evitar erros.

victor-nasc commented on docs/tutorials/installing-python.md:
````diff
@@ -0,0 +1,138 @@
+#### Instalação de um pacote com anaconda
+A fim de instalar um pacote no ambiente anaconda, basta rodar:
+```bash
+conda install <package_name>
````
TAmbem da pra instalar pacotes via pip
dando `conda install pip` dentro do env ou `conda create -n myenv python=3.7 pip` pra criar já com pip
- `pip install numpy`

victor-nasc reviewed (commented):
achei bem bom. gostei que deu bastante detalhes

J-Linaris committed:
adiciona o pip como possibilidade de instalador de pacotes, além de mudar todos os nomes em inglês, como <package_name> para português, virando <nome_do_pacote>

J-Linaris committed:
Merge branch 'master' of https://github.com/lucasmsobrinho/docs-lab-vision into config-env
